### PR TITLE
php72.extensions.intl: Fix builds on Darwin.

### PR DESCRIPTION
**Pull request by drupol**
It seems that it is not possible to build PHP with Intl extension on Darwin platform.

See: https://github.com/loophp/nix-shell/runs/5142961981?check_suite_focus=true

This patch should fix the issue.

PS: I don't have any Mac to test this.

### CODE REVIEW

File: pkgs/package-overrides.nix [hunk 1]
```diff
@@ -95,6 +95,9 @@ in
                   patch "$out" < ${if lib.versionOlder prev.php.version "7.0" then ./intl-icu-patch-5.6-compat.patch else ./intl-icu-patch-7.0-compat.patch}
                 '';
               })
+            ] ++
+            lib.optionals (prev.php.version == "7.2") [
```
Comment (jtojnar) [inline pkgs/package-overrides.nix:99]: Again, version is something like `"7.2.34"` so this will not match anything.

Comment (drupol) [reply pkgs/package-overrides.nix:99]: Indeed ! Will fix accordingly.

File: pkgs/package-overrides.nix [hunk 2]
```diff
@@ -95,6 +95,9 @@ in
...
+              ./bug76826.patch
```
[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/package-overrides.nix:100]: This PR is fine, as long as the commit precedes this one. But whatever you prefer.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -9,10 +9,11 @@ on:
 
 jobs:
   build:
-    name: 'PHP ${{ matrix.php.major }}.${{ matrix.php.minor }}'
-    runs-on: ubuntu-20.04
+    name: 'PHP ${{ matrix.php.major }}.${{ matrix.php.minor }} on ${{ matrix.operating-system }}'
+    runs-on: ${{ matrix.operating-system }}
     strategy:
       matrix:
+        operating-system: [ubuntu-latest, macOS-latest]
```
Comment (jtojnar) [inline .github/workflows/build.yaml:16]: Please use itemized lists:

```suggestion
        operating-system:
          - ubuntu-latest
          - macOS-latest
```

File: pkgs/package-overrides.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/package-overrides.nix:102]: Ok let's tackle all the patches in their own directory once this is done. Ok ?

File: pkgs/package-overrides.nix [hunk 4]
```diff
@@ -96,7 +96,7 @@ in
                 '';
               })
             ] ++
-            lib.optionals (prev.php.version == "7.2") [
+            lib.optionals (lib.versionOlder prev.php.version == "7.1" && lib.versionAtLeast prev.php.version == "7.3") [
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/package-overrides.nix]: Done.

**Comment by drupol** [comment]
@jtojnar How should we fix this? https://github.com/fossar/nix-phps/runs/5147982211?check_suite_focus=true

File: pkgs/package-overrides.nix [hunk 5]
```diff
@@ -95,6 +95,9 @@ in
                   patch "$out" < ${if lib.versionOlder prev.php.version "7.0" then ./intl-icu-patch-5.6-compat.patch else ./intl-icu-patch-7.0-compat.patch}
                 '';
               })
+            ] ++
+            lib.optionals (lib.versions.MajorMinor prev.php.version == "7.2") [
+              ./bug76826.patch
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/package-overrides.nix:102]: Done!

**Comment by jtojnar** [comment]
Searching for “shopt: inherit_errexit” found https://github.com/NixOS/nixpkgs/issues/155706

**Comment by jtojnar** [comment]
I would probably go with using nixpkgs from the flake.

**Comment by jtojnar** [comment]
Or maybe convert the nix-shell invocations to the new nix command as well.

**Comment by drupol** [comment]
@jtojnar Do you have a piece of advice on how we should fix failing builds?

**Pull request closed without merging**